crates/mir/src/emit/container.rs

```diff
@@ -5,41 +5,43 @@ use rokugo_common::color::ColoredDisplay;
 use super::op_code::{MirInstruction, MirInstructionData, MirInstructionMeta, MirOpCode};
 
 #[derive(Debug)]
-pub struct MirContent {
+pub struct MirContainer {
     pub(super) data: Vec<u8>,
 }
 
-impl MirContent {
-    pub fn iter(&self) -> MirContentIterator {
-        MirContentIterator {
+impl MirContainer {
+    pub fn iter(&self) -> MirContainerIterator {
+        MirContainerIterator {
             content: self,
             index: 0,
         }
     }
 
-    pub(super) unsafe fn emit_native_bytes<T>(&mut self, value: T) {
-        self.data.extend_from_slice(std::slice::from_raw_parts(
-            &value as *const _ as *const u8,
-            std::mem::size_of::<T>(),
-        ));
+    pub(super) fn emit_native_bytes<T>(&mut self, value: T) {
+        // SAFETY: This is safe, because `T` lifetime does not exceed the lifetime of slice, and this slice is only used
+        // for copying bytes.
+        let slice = unsafe {
+            std::slice::from_raw_parts(&value as *const _ as *const u8, std::mem::size_of::<T>())
+        };
+        self.data.extend_from_slice(slice);
```

Comment:
Bad idea. This API does not ensure that `T` is a plain old data type without any pointers inside. Deserializing such a value from disk would result in **instant** Undefined Behavior, by loading in pointers which are most definitely not valid.

The entirety of the [`bytemuck`] crate means to make this into a safe operation. Please use it instead of mucking with bytes without guard rails like this.

[`bytemuck`]: https://lib.rs/crates/bytemuck